ta/Makefile

```diff
@@ -5,6 +5,8 @@ ifeq ($O,)
 $(error output path should be specified when calling this makefile)
 endif
 
+LDFLAGS=
```

Comment:
This is not consistent with the other changes in this patch: you want to be able to override `CC`, `CPP`, `LD` etc. via the environment, but you don't want the environment to influence `LDFLAGS`. This does not sound good.
Also, this introduces a different behavior between environment and command line (`LDFLAGS=foo make` and `make LDFLAGS=foo` will not do the same), which I don't really like.
Can't you just clear `LDFLAGS` before running this Makefile?